sdk/search/Azure.Search.Documents/src/Models/SynonymMap.cs

```diff
@@ -0,0 +1,78 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using Azure.Core;
+
+namespace Azure.Search.Documents.Models
+{
+    [CodeGenSchema("SynonymMap")]
+    public partial class SynonymMap
+    {
+        private const string DefaultFormat = "soln";
+        private const string NewLine = "\n";
+
+        // TODO: Replace constructor and read-only properties when https://github.com/Azure/autorest.csharp/issues/554 is fixed.
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SynonymMap"/> class.
+        /// </summary>
+        /// <param name="name">The name of the synonym map.</param>
+        /// <param name="synonym">The synonym to define.</param>
+        /// <param name="format">The format of the synonym map. Currently, only "soln" is supported and is the default value.</param>
+        /// <exception cref="ArgumentException"><paramref name="name"/>, <paramref name="synonym"/>, or <paramref name="format"/> is an empty string.</exception>
+        /// <exception cref="ArgumentNullException"><paramref name="name"/>, <paramref name="synonym"/>, or <paramref name="format"/> is null.</exception>
+        public SynonymMap(string name, string synonym, string format = DefaultFormat)
```

Comment:
Is there any reason to support `format` now?  We could just add an overload once it is allowed in the future.

Reply:
I have a separate bug for this already.